### PR TITLE
Split SAI API

### PR DESCRIPTION
**Pull request by marian-pritsak**
Make APIs compatible with SONiC bulk infra

Signed-off-by: Marian Pritsak <marianp@mellanox.com>

### CODE REVIEW

**Review by chrispsommers — CHANGES_REQUESTED** [review]
Changing the name of the DASH apis requires that sai_api_query() be updated accordingly. The error in the CI run https://github.com/marian-pritsak/DASH-1/runs/7963289127?check_suite_focus=true#step:7:365 shows a mismatch between sai API datatypes and API enums.

The SAI API code is here: https://github.com/Azure/DASH/blob/20b39d325257f080c77f05302cc4198cbf8c29f9/dash-pipeline/SAI/templates/utils.cpp.j2#L254. It needs to be refactored to handle the split APIs. It should only take a few minutes. It'd be better if this were auto-generated (and located in a different file, e.g. sai_api.cpp.j2) so it can track the P4 code automatically.

**Comment by marian-pritsak** [comment]
@chrispsommers Are you familiar with this error?
```
OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "./vnet_out": stat ./vnet_out: no such file or directory: unknown
```
Was something recently changed in the directory structure?

**Comment by chrispsommers** [comment]
> @chrispsommers Are you familiar with this error?
> 
> ```
> OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "./vnet_out": stat ./vnet_out: no such file or directory: unknown
> ```
> 
> Was something recently changed in the directory structure?

Hi Marian, there have been no changes to the structure. I've seen this error when I executed some variation on `make clean` which deleted the `dash-pipeline/tests/libsai/*` directories. Those get built via `make p4 sai test` or of course, `make all`. Can you provide the context in which this occurred, i.e. was it in your workspace or CI pipeline? What branch? What sequence did you follow? I'll be glad to help.

**Comment by chrispsommers** [comment]
@marian-pritsak Two Makefile suggestions came out of the WG meeting today from others:
Guohan suggested `.ONESHELL = -e`
James Lo suggested `set -e` in begging of the script.

File: dash-pipeline/SAI/templates/utils.cpp.j2 [hunk 1]
```diff
@@ -245,9 +245,9 @@ sai_switch_api_t sai_switch_api_impl = {
 
 
 /* TODO [cs] This should be auto-generated or part of per-API, auto-generated include file */
```
Comment (chrispsommers) [inline dash-pipeline/SAI/templates/utils.cpp.j2:247]: This comment can be deleted.

File: dash-pipeline/SAI/templates/utils.cpp.j2 [hunk 2]
```diff
@@ -245,9 +245,9 @@ sai_switch_api_t sai_switch_api_impl = {
...
-extern sai_dash_api_t sai_dash_api_impl;
-extern sai_dash_vnet_api_t sai_dash_vnet_api_impl;
-extern sai_dash_acl_api_t sai_dash_acl_api_impl;
+{% for api in api_names %}
+extern sai_{{ api }}_api_t sai_{{ api }}_api_impl;
+{% endfor %}
 
 
 /* TODO [cs] This should be auto-generated */
```
Comment (chrispsommers) [inline dash-pipeline/SAI/templates/utils.cpp.j2:253]: This comment can be deleted.

**Review by chrispsommers — CHANGES_REQUESTED** [review]
I think that the source branch needs to by synced with main branch to ensure the latest PTF tests continue to pass with current P4 code and libsai. It's 23 commits behind main. In particular, the latest [vnet test](https://github.com/Azure/DASH/blob/publish-dockers-to-acr/dash-pipeline/tests/saithrift/ptf/vnet/test_saithrift_vnet.py) now sends legitimate a VxLAN packet instead of "simple UDP" packet so we need to ensure this PR doesn't break anything.

Autogenerated libsai code looks good, I verified it builds and runs.

**Comment by marian-pritsak** [comment]
@chrispsommers, I moved commit on top of the origin/main

**Comment by chrispsommers** [comment]
Hi @marian-pritsak merged code looks good. Would you mind deleting those minor comments in the templates, they're no longer needed and might confuse someone. Thanks.

File: dash-pipeline/SAI/templates/utils.cpp.j2 (hunk 1 above)
Comment (marian-pritsak) [reply dash-pipeline/SAI/templates/utils.cpp.j2]: Done

File: dash-pipeline/SAI/templates/utils.cpp.j2 (hunk 2 above)
Comment (marian-pritsak) [reply dash-pipeline/SAI/templates/utils.cpp.j2]: Done

**Review by chrispsommers — APPROVED** [review]

**Comment by chrispsommers** [comment]
LGTM, let's merge it after latest CI passes. Thanks for making the requested changes!

**Pull request merged**